source/coap_connection_handler.c

```diff
@@ -11,9 +11,16 @@
 #include "socket_api.h"
 #include "net_interface.h"
 #include "eventOS_event_timer.h"
+#include "coap_service_api_internal.h"
 
 #define TRACE_GROUP "ThCH"
 
+typedef enum session_state_e {
+    SECURE_SESSION_HANDSHAKE_ONGOING = 0,
+    SECURE_SESSION_OK,
+    SECURE_SESSION_ALERT_SENT
```

Comment:
should alert sent be name SECURE_SESSION_CLOSED
and the definition be more generic